### DISCUSSION TITLE
Regarding ifm3d logging file.

### DISCUSSION THREAD

**Discussion by Astronomia-maker** [ifm/ifm3d#424] [Q&A]
Hi,

I have my application running on VPU800 with two ifm O3R 222 camera using python as base docker image. I am logging and creating my own log file name test.log. But i am also getting IFM3D_invaliduser and maynmore logging from camera itself on VPU800 which is making storage full. 

I have used o3r.get() to get configuration and i found 
`},
                "log": {
                    "components": {},
                    "level": "warn",
                    "storage": "volatile"
                },` 
Can i set level:- none then would it work or not or how can i resolve this issue?

My software stack version:-
ifm3dpy==1.1.1
python==3.10

**Comment by graugans** [comment]
> I have used o3r.get() to get configuration and i found `}, "log": { "components": {}, "level": "warn", "storage": "volatile" },` Can i set level:- none then would it work or not or how can i resolve this issue?


This will change the logs of the firmware running on the OVP8xx it does not impact the logging of ifm3d

**Reply by graugans** [reply]
Please find in the CHANGELOG for [1.3.2](https://github.com/ifm/ifm3d/blob/main/ChangeLog.md#132---2023-06-16) the possibility to define your own logger instead of the previously used GLog framework. The Python Bindings got added in [1.4.1](https://github.com/ifm/ifm3d/blob/main/ChangeLog.md#141---2023-08-31) More Details can be found here.

I guess your use-case is solved by updating the ifm3dpy version to 1.3.3 because the logs are dumped to stdout instead of writing them to a file.

**Reply by graugans** [reply]
In case you want to stick with 1.1.1 you can try this:

```
# Write the IFM3D logs to the stderr instead to a file
export GLOG_logtostderr=1

# Do not log anything below fatal
export GLOG_minloglevel=3
```

This will instruct GLog to not write any files and log to `stderr` instead of files.  See the [Glog](https://github.com/google/glog#setting-flags) docu for details. This will no longer work for ifm3d versions > 1.3.2

**Comment by Astronomia-maker** [comment]
Thank you @graugans for informative and detailed answer.
